api/routes/index.js

```diff
@@ -183,21 +186,23 @@ router.put('/coop', isAuthenticated, async (req, res) => {
     profile_pic,
   ];
 
-  var queryTags = await db.query(
-    'SELECT tag_name FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id = $1',
-    [coopId]
-  );
-  var listOfTagsDatabase = getArrayOfTags(queryTags);
-
-  deleteArray = diffArray(listOfTagsDatabase, tags);
-  addArray = diffArray(tags, listOfTagsDatabase);
-
   try {
     await db.query(updateQueryText, updateQueryValues);
-    await db.query(
-      `DELETE FROM coop_tags WHERE coop_id = $1 AND tag_id IN (SELECT id from tags WHERE tag_name = ANY ($2));`,
-      [coopId, deleteArray]
+
+    await db.query('DELETE FROM coop_tags WHERE coop_id = $1', [coopId]);
+
+    var namesToIds = await db.query(
+      'SELECT id from tags WHERE tag_name = ANY ($1)',
+      [tags]
     );
+    namesToIds = namesToIds.rows.map(dict => [coopId, dict['id']]);
+    if (namesToIds.length > 0) {
+      const queryInsertTags = format(
+        `INSERT INTO coop_tags (coop_id, tag_id) VALUES %L`,
+        namesToIds
+      );
+      await db.query(queryInsertTags);
+    }
```

Comment:
my bad i should've mentioned this earlier, but let's make all these operations atomic (either all succeed or all fail) with one [transaction](https://node-postgres.com/features/transactions) like this
```javascript
try {
  await db.query('BEGIN');
  // your SQL updates, deletes, and inserts (lines 190 to 205)
  await db.query('COMMIT');
  res.send(`Successfully updated coop ${coopId}`);
} catch (err) {
  await db.query('ROLLBACK');
  console.log(err.stack);
}